### DISCUSSION TITLE
exit loop based on if/else condition

### DISCUSSION THREAD

**Discussion by phileinsophos** [cypress-io/cypress#15576] [Component Testing] [answered]
```
/// <reference types='cypress' />


it("Visiting Google",function(){
    var webUrl = 'https://html5test.com/'
    cy.visit(webUrl)
    var iterator = 1
    while(iterator < 5)
    {
        const $select_element = cy.get('h1 > em')
        if($select_element.contains('abcd'))
        {
            cy.log("string found")
        }
        else
        {
            cy.log('string not found')
            break
        }
        iterator += 1
    }
    })
```

How do I successfully run this code block.
Currently when if condition is true, it executes if block, but when the condition fails, it raises AssertionError. I want the program to goto else block and execute statements inside else block

**Comment by bahmutov** [comment, marked as answer]
Cypress commands are asynchronous, thus the `while` or `for` loops will always NOT work as you expect them. Instead you need to use recursion. See this recipe https://github.com/cypress-io/cypress-example-recipes/tree/master/examples/testing-dom__page-reloads and implement your logic using https://github.com/bahmutov/cypress-recurse
